nncf/config/schemata/algo/quantization.py

```diff
@@ -0,0 +1,512 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+from nncf.config.definitions import ONLINE_DOCS_ROOT
+from nncf.config.schemata.basic import ARRAY_OF_NUMBERS
+from nncf.config.schemata.basic import ARRAY_OF_STRINGS
+from nncf.config.schemata.basic import BOOLEAN
+from nncf.config.schemata.basic import NUMBER
+from nncf.config.schemata.basic import STRING
+from nncf.config.schemata.basic import annotated_enum
+from nncf.config.schemata.basic import with_attributes
+from nncf.config.schemata.common.compression import BASIC_COMPRESSION_ALGO_SCHEMA
+from nncf.config.schemata.common.compression import COMPRESSION_LR_MULTIPLIER_PROPERTY
+from nncf.config.schemata.common.initialization import BATCHNORM_ADAPTATION_SCHEMA
+from nncf.config.schemata.common.targeting import SCOPING_PROPERTIES
+from nncf.config.schemata.defaults import ACTIVATIONS_QUANT_START_EPOCH
+from nncf.config.schemata.defaults import AUTOQ_EVAL_SUBSET_RATIO
+from nncf.config.schemata.defaults import AUTOQ_WARMUP_ITER_NUMBER
+from nncf.config.schemata.defaults import HAWQ_COMPRESSION_RATIO
+from nncf.config.schemata.defaults import HAWQ_DUMP_INIT_PRECISION_DATA
+from nncf.config.schemata.defaults import HAWQ_ITER_NUMBER
+from nncf.config.schemata.defaults import HAWQ_NUM_DATA_POINTS
+from nncf.config.schemata.defaults import HAWQ_TOLERANCE
+from nncf.config.schemata.defaults import LR_POLY_DURATION_EPOCHS
+from nncf.config.schemata.defaults import MAX_PERCENTILE
+from nncf.config.schemata.defaults import MIN_PERCENTILE
+from nncf.config.schemata.defaults import NUM_INIT_SAMPLES
+from nncf.config.schemata.defaults import PRECISION_INIT_BITWIDTHS
+from nncf.config.schemata.defaults import QUANTIZATION_BITS
+from nncf.config.schemata.defaults import QUANTIZATION_EXPORT_TO_ONNX_STANDARD_OPS
+from nncf.config.schemata.defaults import QUANTIZATION_LOGARITHM_SCALE
+from nncf.config.schemata.defaults import QUANTIZATION_PER_CHANNEL
+from nncf.config.schemata.defaults import QUANTIZATION_PRESET
+from nncf.config.schemata.defaults import QUANTIZE_INPUTS
+from nncf.config.schemata.defaults import QUANTIZE_OUTPUTS
+from nncf.config.schemata.defaults import STAGED_QUANTIZATION_BASE_LR
+from nncf.config.schemata.defaults import STAGED_QUANTIZATION_BASE_WD
+from nncf.config.schemata.defaults import WEIGHTS_QUANT_START_EPOCH
+
+QUANTIZER_MODES = ['symmetric', 'asymmetric']
+
+QUANTIZER_CONFIG_PROPERTIES = {
```

Comment:
@nikita-savelyevv please review the rest of the quantization config except for the precision init part